### PR TITLE
Update tinydir.h

### LINKED ISSUE #20678: tinydir.h is outdated and throws some compile warnings
## Reason
Some struct members in an internal data structure are potentially uninitialized. 

## Design
We can simply set those to nullptr.

## Impact
No more warnings

### PR DESCRIPTION
**Pull request by dschwen**
Closes #20678

### CODE REVIEW

File: framework/contrib/tinydir/include/tinydir.h [hunk 1]
```diff
@@ -25,401 +29,802 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+  } tinydir_file;
+
+  typedef struct tinydir_dir
+  {
+    _tinydir_char_t path[_TINYDIR_PATH_MAX];
+    int has_next;
+    size_t n_files;
 
-	strcpy(dir->path, path);
+    tinydir_file * _files = nullptr;
```
Comment (dschwen) [inline framework/contrib/tinydir/include/tinydir.h:218]: this initialization and the one in l223 fix the warnings

**Review by cticenhour — CHANGES_REQUESTED** [review]
Can you take a look in `MooseUtils.C:36` and remove those ignores I added for GCC and tinydir? If you're fixing all of these up, they should be removed.

**Comment by dschwen** [comment]
Ha, didn't even see those because I was working off an older devel 😊

**Review by cticenhour — APPROVED** [review]
Thanks @dschwen!

**Pull request merged**